### PR TITLE
chakrashim: cleanup inspector code

### PR DESCRIPTION
**Pull request by kfarnung**
* Use cached prop IDs throughout
* Use helpers throughout
* Replace nullptr with JS_INVALID_REFERENCE where appropriate
* Shorten long lines

##### Checklist

- [x] `make -j4 test` (UNIX), or `vcbuild test` (Windows) passes
- [x] commit message follows [commit guidelines](https://github.com/nodejs/node/blob/master/CONTRIBUTING.md#commit-message-guidelines)

##### Affected core subsystem(s)

chakrashim

### CODE REVIEW

File: deps/chakrashim/src/jsrtinspectorhelpers.cc [hunk 1]
```diff
@@ -47,18 +49,15 @@ namespace jsrt {
...
     }
 
-    JsPropertyIdRef propId = nullptr;
-    IfJsErrorRet(JsCreatePropertyId(sourceName, strlen(sourceName), &propId));
-
     bool hasProp;
-    IfJsErrorRet(JsHasProperty(sourceObj, propId, &hasProp));
+    IfJsErrorRet(JsHasProperty(sourceObj, sourceIdRef, &hasProp));
 
     if (!hasProp) {
```
Comment (agarwal-sandeep) [inline deps/chakrashim/src/jsrtinspectorhelpers.cc]: >hasProp [](start = 9, length = 7)

nit: initialize hasProp and isUndefined

**Comment by kfarnung** [comment]
CI: https://ci.nodejs.org/job/chakracore-test-pull-request/70/

File: deps/chakrashim/src/jsrtinspectorhelpers.cc (hunk 1 above)
Comment (kfarnung) [reply deps/chakrashim/src/jsrtinspectorhelpers.cc]: Thanks!

File: deps/chakrashim/src/inspector/java-script-call-frame.cc [hunk 2]
```diff
@@ -88,7 +92,8 @@ int JavaScriptCallFrame::contextId() const {
 
 bool JavaScriptCallFrame::isAtReturn() const {
   int index;
-  if (InspectorHelpers::GetIntProperty(m_callFrame, "index", &index) != JsNoError) {
+  if (jsrt::GetProperty(m_callFrame, jsrt::CachedPropertyIdRef::index,
+                        &index) != JsNoError) {
     assert(false);
     return false;
   }
```
Comment (agarwal-sandeep) [inline deps/chakrashim/src/inspector/java-script-call-frame.cc]: For such pattern should we create a macro IfJsErrorAssertAndRetFalse? will make code clean

File: deps/chakrashim/src/jsrtutils.cc [hunk 3]
```diff
@@ -1018,18 +961,23 @@ StringUtf8::~StringUtf8() {
...
   size_t written = 0;
   JsErrorCode errorCode = JsCopyString(strRef, buffer, len, &written);
+  
   if (errorCode == JsNoError) {
     CHAKRA_ASSERT(len == written);
     buffer[len] = '\0';
     _str = buffer;
     _length = static_cast<int>(len);
   }
+
```
Comment (agarwal-sandeep) [inline deps/chakrashim/src/jsrtutils.cc]: extra spaces? can you run cpplint on all changed files and make sure you didn't introduced new error/warnings

File: deps/chakrashim/src/jsrtutils.cc [hunk 4]
```diff
@@ -37,31 +37,20 @@ JsErrorCode GetProperty(JsValueRef ref,
                         JsValueRef propName,
                         JsValueRef *result) {
   JsPropertyIdRef idRef;
```
Comment (agarwal-sandeep) [inline deps/chakrashim/src/jsrtutils.cc]: >idRef [](start = 18, length = 5)

initialize?

File: deps/chakrashim/src/jsrtutils.cc [hunk 5]
```diff
@@ -75,16 +64,40 @@ JsErrorCode GetProperty(JsValueRef ref,
 
 JsErrorCode GetProperty(JsValueRef ref,
                         JsPropertyIdRef propId,
+                        bool *boolValue) {
+  JsValueRef value;
```
Comment (agarwal-sandeep) [inline deps/chakrashim/src/jsrtutils.cc]: >value [](start = 13, length = 5)

initialize, please check other places as well

File: deps/chakrashim/src/jsrtutils.cc [hunk 6]
```diff
@@ -97,32 +110,68 @@ JsErrorCode SetProperty(JsValueRef ref,
...
+  IfJsErrorRet(JsSetProperty(ref, idRef, propValue, false));
 
-  return error;
+  return JsNoError;
+}
+
+JsErrorCode SetProperty(JsValueRef ref,
+                        JsPropertyIdRef propId,
+                        bool boolValue) {
+  JsValueRef boolVal = nullptr;
```
Comment (agarwal-sandeep) [inline deps/chakrashim/src/jsrtutils.cc]: >nullptr [](start = 23, length = 7)

Not nullptr, use JS_INVALID_REFERENCE

File: deps/chakrashim/src/inspector/java-script-call-frame.cc (hunk 2 above)
Comment (kfarnung) [reply deps/chakrashim/src/inspector/java-script-call-frame.cc]: I actually just switched all these to CHAKRA_VERIFY_NOERROR which was the original intention, but at that point the code was separated and that macro wasn't available.

**Comment by kfarnung** [comment]
New CI: https://ci.nodejs.org/job/chakracore-test-pull-request/71/

**Comment by agarwal-sandeep** [comment]
:shipit:

**Comment by kfarnung** [comment]
@kunalspathak Could you take a look at the vcbuild.bat changes?

**Pull request merged**